recipes/el-secretario-org

```diff
@@ -0,0 +1,4 @@
+(el-secretario-org
+ :fetcher git
+ :url "https://git.sr.ht/~zetagon/el-secretario"
+ :files ("el-secretario-org.el" "el-secretario-space.el"))
```

Comment:
`el-secretario-org.el` with byte-compile using Emacs 27.1:
```
el-secretario-org.el:83:1:Warning: Unused lexical variable `keymap'
el-secretario-org.el:213:1:Warning: global/dynamic var `date' lacks a prefix
```

`el-secretario-space.el` with byte-compile using Emacs 27.1:
```
el-secretario-space.el:88:10:Warning: called-interactively-p called with 0
    arguments, but requires 1
```
`el-secretario-space.el` with package-lint 20211122.1152:
```
6 issues found:
41:0: error: "el-secretario-space-increment-percentage" doesn't start with package's prefix "el-secretario-org".
45:0: error: "el-secretario-space--increment" doesn't start with package's prefix "el-secretario-org".
70:0: error: "el-secretario-space--reset" doesn't start with package's prefix "el-secretario-org".
74:0: error: "el-secretario-space-reschedule" doesn't start with package's prefix "el-secretario-org".
82:0: error: "el-secretario-space-schedule-and-reset" doesn't start with package's prefix "el-secretario-org".
95:0: error: "el-secretario-space-compare-le" doesn't start with package's prefix "el-secretario-org".
```

Reply:
> el-secretario-org.el:213:1:Warning: global/dynamic var `date' lacks a prefix
This is a hack around some functionality that requires a `date` variable to exist with dynamic scoping.

> el-secretario-space.el:88:10:Warning: called-interactively-p called with 0
    arguments, but requires 1

I'm honestly not sure why this becomes a false positive. The definition looks like this:`(defun called-interactively-p (&optional kind)`.
> 41:0: error: "el-secretario-space-increment-percentage" doesn't start with package's prefix "el-secretario-org".

I don't understand this. Shouldn't the prefix be of the filename? That has been the case for the main package.

Reply:
> I'm honestly not sure why this becomes a false positive.

That's interesting.  I'm not sure why the `&optional` isn't registering (neither by the compiler or by the Emacs help) but if you check out the documentation there seems to be a workaround:

> > Instead of using this function, it is cleaner and more reliable to give your
> > function an extra optional argument whose ‘interactive’ spec specifies
> > non-nil unconditionally ("p" is a good way to do this), or via
> > (not (or executing-kbd-macro noninteractive)).


> Shouldn't the prefix be of the filename? That has been the case for the main package.

The prefix should be of the package-name, and it's also best to prefix all of the files inside a given package with the package name too (or else someone might innocently add e.g. an `el-secretario-space` package and cause a namespace collision).


Also, as I mentioned above, please double-check the checkdoc warnings.  They were not all false positives as I originally assumed.  That's my oversight -- apologies.

Reply:
I implemented a workaround and renamed the file.

Reply:
Thanks for accommodating!  I think only this one remains:

`el-secretario-org.el` with byte-compile using Emacs 27.1:
```
el-secretario-org.el:272:1:Warning: global/dynamic var `date' lacks a prefix
```
and that seems to be an easy fix -- you don't need to `defvar` date as far as I can tell.

Reply:
Hm, I was sure it was needed but it looks like it works without it. Anyways, now it's fixed.

Reply:
Actually no, I haven't had time to test yet, but I think the `(defvar date)` is needed to make `date` dynamically scoped. `org-agenda-get-deadlines` requires that the caller sets `date` dyamically, and because I have `-*- lexical-binding: t; -*-` the `let` binding will be lexically scope so `org-agenda-get-deadlines` won't see the binding.

Reply:
I tested it now, and yes this leads to an error so I will re/add it with `with-no-warnings` to supress the warning.

Reply:
Ahh I see -- that's the same approach org-agenda takes (`with-no-warnings`).  I just learned something new. :)